### PR TITLE
Fix/118334 clear shared capacity

### PR DESCRIPTION
**Pull request by paulmskim**
For ticket: https://central.tri.be/issues/ 118334

### CODE REVIEW

File: src/modules/data/blocks/ticket/__tests__/sagas.test.js [hunk 1]
```diff
@@ -395,6 +407,41 @@ describe( 'Ticket Block sagas', () => {
...
+		it( 'should remove tickets block', () => {
+			const gen = cloneableGenerator( sagas.removeTicketsBlock )();
+			expect( gen.next().value ).toEqual(
+				select( selectors.getSharedTicketsCount )
+			);
+
+			const clone1 = gen.clone();
+			expect( clone1.next( 2 ).done ).toEqual( true );
+
+			const clone2 = gen.clone();
```
Comment (nealfennimore) [inline src/modules/data/blocks/ticket/__tests__/sagas.test.js]: Why the clones? Is there an issue somewhere getting this to pass without them?

File: src/modules/data/blocks/ticket/__tests__/sagas.test.js [hunk 2]
```diff
@@ -395,6 +407,41 @@ describe( 'Ticket Block sagas', () => {
...
+			expect( JSON.stringify( clone2.next( 0 ).value ) ).toEqual(
```
Comment (nealfennimore) [inline src/modules/data/blocks/ticket/__tests__/sagas.test.js]: I'd just do a snapshot here @paulmskim. It's a lot of work in some cases to model the state correct, and make sure separate calls are using the same functions.

File: src/modules/data/blocks/ticket/sagas.js [hunk 3]
```diff
@@ -98,6 +98,22 @@ export function* setTicketsInitialState( action ) {
 	yield put( actions.setTicketsProvider( provider || defaultProvider ) );
 }
 
+export function* removeTicketsBlock() {
```
Comment (nealfennimore) [inline src/modules/data/blocks/ticket/sagas.js]: Might be misunderstanding this, but it sort of looks like we're resetting the ticket. Are we really removing it?

**Comment by nealfennimore** [comment]
Just some general questions other looks :+1:

File: src/modules/data/blocks/ticket/__tests__/sagas.test.js (hunk 1 above)
Comment (paulmskim) [reply src/modules/data/blocks/ticket/__tests__/sagas.test.js]: there's an if statement at the start depending on how many shared tickets we have, so testing both cases.

File: src/modules/data/blocks/ticket/sagas.js (hunk 3 above)
Comment (paulmskim) [reply src/modules/data/blocks/ticket/sagas.js]: I can rename to `resetTicketsBlock` instead, that'll be less confusing.

File: src/modules/data/blocks/ticket/__tests__/sagas.test.js (hunk 1 above)
Comment (nealfennimore) [reply src/modules/data/blocks/ticket/__tests__/sagas.test.js]: Can you make a separate `it` for that please? :)

Comment (paulmskim) [reply src/modules/data/blocks/ticket/__tests__/sagas.test.js]: will do

**Comment by paulmskim** [comment]
good calls, thanks @nealfennimore . everything is updated.

**Comment by nealfennimore** [comment]
Thanks!

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for *G19.01*, but this was pull requested against *release/M19.01*.

**Pull request merged**